### PR TITLE
Fix readuntil work 

### LINKED ISSUE #6701: Error in StreamReader#readuntil
### Describe the bug

Hi there, it looks like StreamReader#readuntil function [here](https://github.com/aio-libs/aiohttp/blob/3.8/aiohttp/streams.py#L328) has an error.

My original use case was to read CSV files directly from ClientResponse line-by-line when the line terminator is `\r\n`.
The example I've provided is very similar.

When the separator has more than one symbol, it splits lines in a strange way.
For instance, when separator is `b'ab'`, it splits as follows:
```
b'11a'
b'b\n22a'
b'b\n33a'
b'b'
```
e.g. at the middle of the separator.

I think that this part wasn't updated at the time when it was `readline` function with only `\n` (one symbol length) separator available.

Probably this line should be:
```python
ichar = self._buffer[0].find(separator, offset) + len(separator) 
```

so that it will contain the whole separator instead of 'splitting' it.

### To Reproduce

Run this thing https://gist.github.com/vitaliihonta/b9be77deb0dfea1e9a411e695e6f15c9

### Expected behavior

Each line should contain the whole separator
```
b'11ab'
b'\n22ab'
b'\n33ab'
```

### Logs/tracebacks

```python-traceback
None
```

### Python Version

```console
Python 3.8.9
```

### aiohttp Version

```console
3.8.1
```

### multidict Version

```console
Name: multidict
Version: 6.0.2
```

### yarl Version

```console
Name: yarl
Version: 1.7.2
```

### OS

macOS

### Related component

Client

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

### PR DESCRIPTION
**Pull request by Tingaev**
## What do these changes do?

Fix work `readuntil()` with separator length more one character

## Are there changes in behavior for the user?
No

## Related issue number
Fixes #6701

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: tests/test_streams.py [hunk 1]
```diff
@@ -377,23 +377,24 @@ async def test_readline_exception(self) -> None:
...
         stream.feed_data(b"chunk1 ")
-        read_task = loop.create_task(stream.readuntil(b"*"))
+        read_task = loop.create_task(stream.readuntil(separator))
 
         def cb():
             stream.feed_data(b"chunk2 ")
-            stream.feed_data(b"chunk3 ")
-            stream.feed_data(b"* chunk4")
+            stream.feed_data(b"chunk3 " + separator)
+            stream.feed_data(b" chunk4")
```
Comment (Dreamsorcerer) [inline tests/test_streams.py]: This one seems to have changed. Deliberate, or should we prefix with `separator`?

Comment (Tingaev) [reply tests/test_streams.py]: I'll do it like before
What do you think about the other tests?
Should I use pytest's parametrization decorator in other tests?

Comment (Dreamsorcerer) [reply tests/test_streams.py]: Would probably be nice, just to have that extra coverage, but it's up to you.

File: tests/test_streams.py [hunk 2]
```diff
@@ -377,116 +377,127 @@ async def test_readline_exception(self) -> None:
...
         line = await read_task
-        assert b"chunk1 chunk2 chunk3 *" == line
+        assert b"chunk1 chunk2 chunk3 " + separator == line
 
         stream.feed_eof()
         data = await stream.read()
         assert b" chunk4" == data
 
-    async def test_readuntil_limit_with_existing_data(self) -> None:
+    @pytest.mark.parametrize("separator", [b"*", b"**"])
```
Comment (Dreamsorcerer) [inline tests/test_streams.py]: Thanks, looks good, but I'd reuse the separators used originally in each test (e.g. "&" and "&&" in this test).

File: tests/test_streams.py [hunk 3]
```diff
@@ -377,116 +377,127 @@ async def test_readline_exception(self) -> None:
...
+    @pytest.mark.parametrize("separator", [b"&", b"&&"])
+    async def test_readuntil_limit(self, separator: bytes) -> None:
         loop = asyncio.get_event_loop()
         # Read one chunk. StreamReaders are fed with data after
         # their 'readuntil' methods are called.
         stream = self._make_one(limit=4)
 
         def cb():
             stream.feed_data(b"chunk1")
-            stream.feed_data(b"chunk2$")
```
Comment (Dreamsorcerer) [inline tests/test_streams.py:428]: I meant, look at each test. :P
This one was `$`. Every test appears to use different characters, which may help get a more thorough testing.

**Comment by Tingaev** [comment]
@Dreamsorcerer ping

**Pull request merged**